src/promptflow/promptflow/_utils/exception_utils.py

```diff
@@ -198,55 +198,48 @@ def build_debug_info(self, ex: Exception):
             "innerException": inner_exception,
         }
 
-    def to_dict(self, *, include_debug_info=False):
-        """Return a dict representation of the exception.
+    @property
+    def error_codes(self):
+        """The hierarchy of the error codes.
 
-        This dict specification corresponds to the specification of the Microsoft API Guidelines:
+        We follow the "Microsoft REST API Guidelines" to define error codes in a hierarchy style.
+        See the below link for details:
         https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#7102-error-condition-responses
 
-        Note that this dict representation the "error" field in the response body of the API.
-        The whole error response is then populated in another place outside of this class.
+        For general exceptions, the hierarchy should be:
+        ["SystemError", {exception type name}]
         """
-        if isinstance(self._ex, JsonSerializedPromptflowException):
-            return self._ex.to_dict(include_debug_info=include_debug_info)
-
-        # Otherwise, return general dict representation of the exception.
-        result = {
-            "code": infer_error_code_from_class(SystemErrorException),
-            "message": str(self._ex),
-            "messageFormat": "",
-            "messageParameters": {},
-            "innerError": {
-                "code": self._ex.__class__.__name__,
-                "innerError": None,
-            },
-        }
-        if include_debug_info:
-            result["debugInfo"] = self.debug_info
-
-        return result
-
+        error_codes = [infer_error_code_from_class(SystemErrorException), self._ex.__class__.__name__]
+        return error_codes
```

Comment:
Don't need the `error_code` variable here. Let's merge these two lines into one to return the list directly.

Reply:
Done.